public/app/features/explore/QueryField.tsx

```diff
@@ -116,6 +117,7 @@ export interface TypeaheadFieldState {
   typeaheadPrefix: string;
   typeaheadText: string;
   value: Value;
+  placeholderIndices: number[];
```

Comment:
I don't think the `placeholderIndices` need to be part of the component state since they do no affect anything inside `render()`. You can make them a local component property instead so that updating them will not cause an additional render update. Also, please file them alphabetically.